warehouse/router_identities.go

```diff
@@ -22,15 +21,13 @@ import (
 )
 
 var (
-	shouldPopulateHistoricIdentities            bool
 	populatingHistoricIdentitiesProgressMap     map[string]bool
 	populatingHistoricIdentitiesProgressMapLock sync.RWMutex
 	populatedHistoricIdentitiesMap              map[string]bool
 	populatedHistoricIdentitiesMapLock          sync.RWMutex
 )
 
-func Init2() {
-	config.RegisterBoolConfigVariable(false, &shouldPopulateHistoricIdentities, false, "Warehouse.populateHistoricIdentities")
+func init() {
```

Comment:
Nice. Maybe since we're making changes here we can take advantage of that and simplify some of the code that is using these maps. For example we could store a `struct{}` instead of a `bool` value and we could have functions like `isDestHistoricIdentitiesPopulated` rewritten as such:

```go
func isDestHistoricIdentitiesPopulated(warehouse model.Warehouse) bool {
	populatedHistoricIdentitiesMapLock.RLock()
	_, ok := populatedHistoricIdentitiesMap[uniqueWarehouseNamespaceString(warehouse)]
	populatedHistoricIdentitiesMapLock.RUnlock()
	return ok
}
```

And:

```go
func setDestHistoricIdentitiesPopulated(warehouse model.Warehouse) {
	populatedHistoricIdentitiesMapLock.Lock()
	populatedHistoricIdentitiesMap[uniqueWarehouseNamespaceString(warehouse)] = struct{}{}
	populatedHistoricIdentitiesMapLock.Unlock()
}

func setDestHistoricIdentitiesPopulateInProgress(warehouse model.Warehouse, starting bool) {
	populatingHistoricIdentitiesProgressMapLock.Lock()
	defer populatingHistoricIdentitiesProgressMapLock.Unlock()
	if starting {
		populatingHistoricIdentitiesProgressMap[uniqueWarehouseNamespaceString(warehouse)] = struct{}{}
	} else {
		delete(populatingHistoricIdentitiesProgressMap, uniqueWarehouseNamespaceString(warehouse))
	}
}
```

And so on...